File: My Bank App/KeeperTableViewCell.m
//
//  KeeperTableViewCell.m
//  My Bank App
//
//  Created by Arthur Walasek on 9/23/14.
//  Copyright (c) 2014 Callpod, Inc. All rights reserved.
//

#import "KeeperTableViewCell.h"
#import <KeeperExtensionSDK/KeeperExtensionSDK.h>

@implementation KeeperTableViewCell 

- (void)awakeFromNib {
    // Initialization code
    [self.keeperLockBtn setHidden:![[KeeperSDK sharedExtension] isAppExtensionAvailable]];
}

- (void)setUpCell
{
    [self.txtText setReturnKeyType:self.returnKeyType];
    self.txtText.delegate = self.delegate;
    
    // Take this away to show the Keeper lock within the Text Edit fields
    [self.keeperLockBtn setHidden:TRUE];
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

- (IBAction)keeperLockAction:(id)sender {
    if (self.delegate) {
        if ([self.delegate respondsToSelector:@selector(keeperLockAction)]) {
            [self.delegate performSelector:@selector(keeperLockAction)];
        }
    }
}

@end

File: My Bank App Swift/KeeperTableViewCell.swift
//
//  KeeperTableViewCell.swift
//  My Bank App
//
//  Created by Jesse Gatt on 10/13/14.
//  Copyright (c) Keeper Security Inc. All rights reserved.
//

import UIKit
import KeeperExtensionSDK

class KeeperTableViewCell: UITableViewCell {

    @IBOutlet weak var txtText: UITextField!
    @IBOutlet weak var keeperLockBtn: UIButton!
    var delegate:AnyObject!
    var returnKeyType:UIReturnKeyType!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
        #if USE_KEEPER_TEXTFIELD_ICON
            keeperLockBtn.hidden = !(KeeperSDK.sharedExtension().isAppExtensionAvailable())
        #endif
    }
    
    
    func setUpCell() {
        txtText.returnKeyType = returnKeyType
        txtText.delegate = delegate as? UITextFieldDelegate
        #if USE_KEEPER_TEXTFIELD_ICON
            keeperLockBtn.hidden = true
        #endif
    }

    override func setSelected(selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }
    @IBAction func keeperLockAction(sender: AnyObject) {
        
        if delegate.respondsToSelector(Selector("keeperLockAction:")) {
            delegate.keeperLockAction(sender)
        }

        
    }

}
